indico/modules/events/editing/models/editable.py

```diff
@@ -77,3 +79,16 @@ def __repr__(self):
         return format_repr(self, 'id', 'contribution_id', 'type')
 
     # TODO: state - either a column property referencing the newest revision's state or a normal column
+
+    def can_comment(self, user):
+        return self.editor == user or self.contribution.is_user_associated(user, check_abstract=True)
+
+    @property
+    def editors(self):
+        editors = [self.editor] if self.editor else []
+        if session.user not in editors and self.contribution.is_user_associated(session.user, check_abstract=True):
+            editors.append(session.user)
```

Comment:
this logic feels super weird. also, accessing the session from a model?

Reply:
As to this one, let's discuss it in person.